### PR TITLE
Re-land Android fullscreen support

### PR DESCRIPTION
**Pull request by Piinks**
This re-lands #25785. The original PR was breaking tests internally due to a change in SDK support for navigation bar color.

While refactoring the `SystemChromeStyle`, I noticed that changing the navigation bar color was available at an earlier sdk than we were allowing, so I updated it to match native expectations. (changed from 26 to 21 where earliest available).

This is what broke tests internally on older devices. While it ends up giving the correct behavior for these older versions of Android, it made me realize why we were limiting this in the first place. The customer test has a perfect example:

```dart
SystemChrome.setSystemUIOverlayStyle(SystemUiOverlayStyle.dark.copyWith(
        systemNavigationBarColor: Colors.white,
        systemNavigationBarIconBrightness: Brightness.dark,
        statusBarIconBrightness: Brightness.dark));
```

The ability to change the navigation bar icon colors is only available at sdk 26. So if we were to allow the navigation bar color to be set, you could have white buttons on a white background like in this test. This is not desirable and is probably why it was this way to begin with. I have reverted the change to the way it was previously and documented this.

**This is the diff from the original change: https://github.com/flutter/engine/pull/27018/files/ca8e05f563daa8707b25047b48a3c1ddf7ac9242..0f31f8bf4debcdf977d030871dca41ffe0792ee4** 

---

From Original PR (#25785)

This adds support for Android fullscreen modes.

Framework side: https://github.com/flutter/flutter/pull/81303

Related Issues:
- https://github.com/flutter/flutter/issues/35748
- https://github.com/flutter/flutter/issues/39061
- https://github.com/flutter/flutter/issues/40974
- https://github.com/flutter/flutter/issues/44033
- https://github.com/flutter/flutter/issues/63761
- https://github.com/flutter/flutter/issues/69999

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.
- [x] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

**Comment by Piinks** [comment]
Running through internal testing to make sure this is resolved now. :)

File: shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java [hunk 1]
```diff
@@ -370,15 +370,12 @@ private void setSystemChromeSystemUIOverlayStyle(
     int flags = view.getSystemUiVisibility();
 
     // SYSTEM STATUS BAR -------------------------------------------------------------------
-    // You can't change the color of the system status bar until SDK 21.
+    // You can't change the color of the system status bar until SDK 21, but you can't change the
```
Comment (gaaclarke) [inline shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java:373]: s/but/and

File: shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java [hunk 2]
```diff
@@ -398,15 +399,12 @@ private void setSystemChromeSystemUIOverlayStyle(
     }
 
     // SYSTEM NAVIGATION BAR --------------------------------------------------------------
-    // You can't change the color of the system navigation bar until SDK 21.
+    // You can't change the color of the system navigation bar until SDK 21, but you can't change
```
Comment (gaaclarke) [inline shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java:402]: s/but/and

**Review by blasten — APPROVED** [review]
LGTM

File: shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java [hunk 3]
```diff
@@ -263,49 +368,75 @@ private void setSystemChromeSystemUIOverlayStyle(
...
+    // You can't override the enforced contrast for a transparent status bar until SDK 29.
+    // This overrides the translucent scrim that may be placed behind the bar on SDK 29+ to ensure
+    // contrast is appropriate when using full screen layout modes like Edge to Edge.
+    if (!systemChromeStyle.systemStatusBarContrastEnforced && Build.VERSION.SDK_INT >= 29) {
+      window.setStatusBarContrastEnforced(systemChromeStyle.systemStatusBarContrastEnforced);
+    }
+
+    // SYSTEM NAVIGATION BAR --------------------------------------------------------------
+    // You can't change the color of the system navigation bar until SDK 21, but you can't change
+    // the color of the navigation buttons until SDK 26. We only allow both starting at 23 to
```
Comment (gaaclarke) [inline shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java:403]: Don't you mean `We only allow both starting at 26...`?  That's what the if statement below is saying.

**Review by gaaclarke — CHANGES_REQUESTED** [review]
Requested changes are just the aforementioned comments.

File: shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java (hunk 3 above)
Comment (Piinks) [reply shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java:403]: Whoops! Copy pasta. 😋  Thanks!

Comment (gaaclarke) [reply shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java:403]: 🕵️

**Review by gaaclarke — APPROVED** [review]
lgtm

**Comment by Kotlinovsky** [comment]
Why this pull was reverted?

**Pull request merged**